File: FizzBuzz.Backend/FizzBuzz.Backend.API/Entities/GameSession.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace FizzBuzz.Backend.API.Entities
{
    public class GameSession : BaseEntity
    {
        public Guid GameId { get; set; }
        public string Player { get; set; } = string.Empty;
        public int Duration { get; set; }
        public int ScoreCorrect { get; set; }
        public int ScoreIncorrect { get; set; }
        public DateTime StartedAt { get; set; }
        public DateTime? CompletedAt { get; set; }

        [NotMapped]
        public bool IsCompleted =>
            CompletedAt.HasValue && CompletedAt.Value <= DateTime.UtcNow;

        public Game Game { get; set; } = null!;
        public ICollection<GameAnswer> Answers { get; set; } = [];
    }
}

File: fizzbuzz.frontend/src/types/gameSession.ts
import { GameRule } from './game';

export interface GameSession {
    sessionId: string;
    gameId: string;
    game: string;
    player: string;
    duration: number;
    rules: GameRule[];
    startedAt: string;
    completedAt?: string;
    isCompleted: boolean;
}

export interface SessionState {
    number?: number;
    isExhausted: boolean;
    isCompleted: boolean;
    scoreCorrect: number;
    scoreIncorrect: number;
    totalNumbers: number;
    totalAnswers: number;
    timeRemaining: number;
}

export interface SessionResult {
    game: string;
    player: string;
    duration: number;
    scoreCorrect: number;
    scoreIncorrect: number;
    totalNumbers: number;
    accuracy: number;
    startedAt: string;
    completedAt: string;
}

export interface StartSessionRequest {
    gameId: string;
    player: string;
    duration: number;
}
